doc/developers/auth/jwt.rst

```diff
@@ -6,41 +6,78 @@ JWT (`JSON Web Token <https://tools.ietf.org/html/rfc7519>`_) bearer authenticat
-In the future, CCF will support validating token signatures (currently the responsibility of apps) and automatically refreshing token signing keys from OpenID Provider Configuration endpoints.
+CCF provides support for managing public token signing keys required for validating tokens.
+In the future, CCF will support validating token signatures natively (currently the responsibility of apps).
 
-Storing public token signing keys
----------------------------------
-
-IdPs sign tokens with private keys that are periodically updated.
-Corresponding public keys in the form of certificates are needed to validate the signature of a token and are typically published by IdPs at a well-known location in JWKS (`JSON Web Key Set <https://tools.ietf.org/html/rfc7517>`_) format.
-
-.. note::
-
-    Most IdPs support the OpenID Connect Discovery specification which defines how to discover the location to the  certificates.
-    In short, the IdP issuer URL, for example ``https://login.microsoftonline.com/common/v2.0``, is suffixed with ``/.well-known/openid-configuration`` and the ``jwks_uri`` field of the JSON document at that URL is the location of the JWKS document containing the certificates currently in use. In this example, it would be `<https://login.microsoftonline.com/common/discovery/v2.0/keys>`_.
+Setting up a token issuer with manual key refresh
+-------------------------------------------------
 
 Before adding public token signing keys to a running CCF network, the IdP has to be stored as token issuer with a ``set_jwt_issuer`` proposal:
 
 .. code-block:: bash
 
     $ cat issuer.json
     {
-      "issuer": "https://login.microsoftonline.com/common/v2.0"
+      "issuer": "my-issuer",
+      "auto_refresh": false
     }
     $ python -m ccf.proposal_generator set_jwt_issuer issuer.json
 
+The ``issuer`` field is an arbitrary identifier and can be used during token validation to differentiate between multiple issuers.
+
 Note that ``issuer.json`` has some additional optional fields for more advanced scenarios.
 See :ref:`developers/auth/jwt:Advanced issuer configuration` for details.
 
 After this proposal is accepted, signing keys for an issuer can be updated with a ``set_jwt_public_signing_keys`` proposal:
 
 .. code-block:: bash
 
-    $ curl "https://login.microjsononline.com/common/discovery/v2.0/keys" -o jwks.json
-    $ ISSUER="https://login.microsoftonline.com/common/v2.0"
+    $ ISSUER="my-issuer"
     $ python -m ccf.proposal_generator set_jwt_public_signing_keys $ISSUER jwks.json
 
+``jwks.json`` contains the signing keys as JWKS (`JSON Web Key Set <https://tools.ietf.org/html/rfc7517>`_) document. 
+
+Setting up a token issuer with automatic key refresh
+----------------------------------------------------
+
+Most IdPs follow the OpenID Connect Discovery specification and publish their public token signing keys at a well-known location.
```

Comment:
Is that Identity Providers? If so, I suggest we spell it out, unless we make it a term in the Glossary.

Reply:
IdP is written out in the first sentence of the doc. Is that ok?

Reply:
I'll move it to the Glossary and link it :)